### PR TITLE
Add Compilation and Installation steps in CONTRIBUTING.md

### LINKED ISSUE #2594: Add compilation and instalations steps in CONTRIBUTING.md
How to compile and install Nmstate locally is not described properly. It should be described in a new section in CONTRIBUTING.md

https://github.com/nmstate/nmstate/blob/base/CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by abigailajohn**
This commit introduces a clear and detailed guide for compiling and
installing nmstate from the source. It covers the setup for the
development environment on Linux, installation of Rust's Cargo, and the
necessary steps to install and compile the project.

The guide is designed for new contributors, ensuring an easy setup of
their development environment and facilitating their first
contributions to the project.

Changes include:
- Prerequisite information for different operating systems
- Installation commands for Cargo on various Linux distros
- Steps for cloning the repository and compiling the code
- Instructions for running the compiled 'nmstatectl' tool

Resolves: #2594

### CODE REVIEW

**Comment by liangwen12year** [comment]
https://github.com/nmstate/nmstate/blob/base/CONTRIBUTING.md#write-a-good-commit-message, please make sure that the commit message is wrapped around 72 characters?

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -182,3 +182,39 @@ Do your best to follow the clean code guidelines.
...
+```
+- git clone https://github.com/nmstate/nmstate.git
+- cd nmstate
+```
+
+### Compilation
+Navigate to the rust directory and compile the project:
+```
+- cd rust
+- cargo build
```
Comment (cathay4t) [inline CONTRIBUTING.md]: For compiling the whole project, please use `make` at the top level of  code

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -182,3 +182,39 @@ Do your best to follow the clean code guidelines.
...
+```
+
+### Running the Compiled Program
+After successful compilation, you can run the nmstatectl tool to display the current network state:
+```
+- target/debug/nmstatectl show # To dump the state in json format, use the ‘–json’ flag.
```
Comment (cathay4t) [inline CONTRIBUTING.md]: The flag is `--json`

**Review by cathay4t — CHANGES_REQUESTED** [review]

**Comment by abigailajohn** [comment]
@cathay4t I have made the necessary changes as requested.

**Review by cathay4t — APPROVED** [review]

**Pull request merged**